py/client/pydeephaven/session.py

```diff
@@ -53,32 +56,24 @@ def call_completed(self, exception):
 
     def received_headers(self, headers):
         super().received_headers(headers)
-        if headers:
-            auth_token = bytes(headers.get("authorization")[0], encoding='ascii')
-            if auth_token and auth_token != self._session._auth_token:
-                self._session._auth_token = auth_token
+        header_key = "authorization"
+        try:
+            if headers and header_key in headers:
+                header_value = headers.get(header_key)
+                auth_header_value = bytes(header_value[0], encoding='ascii')
+                if auth_header_value:
+                    self._session._auth_header_value = auth_header_value
+        except Exception as e:
+            logger.exception(f'_DhClientAuthMiddleware.received_headers got headers={headers}')
+            return
+            
 
     def sending_headers(self):
```

Comment:
```suggestion
    def sending_headers(self):
```
Also the line above has space chars on an empty line, I can't get github to let me edit that

Reply:
Done